.github/workflows/auto-deploy-contracts.yaml

```diff
@@ -0,0 +1,42 @@
+name: Auto deploy IPC contracts when changed
+
+on:
+  workflow_dispatch:
+  # TODO: Test only. Remove before PR merge
+  pull_request:
+  push:
+    branches:
+      - main
+    paths:
+      - contracts/**
+
+concurrency:
+  # Only allow one run at a time for this workflow
+  group: auto-deploy-contracts
+  cancel-in-progress: true
+
+jobs:
+  deploy-contracts:
+    runs-on: ubuntu-latest
+    env:
+      RPC_URL: https://calibration.filfox.info/rpc/v1
```

Comment:
Not necessary, but can we move the URL to a secret, and make sure that the subnet deployment automation uses the same URL? Trying to reduce entropy and consolidate to a single configuration spot, in case endpoints become unstable and we need to change them.

Reply:
Note that github secrets do not propagate between github workflows unless you explicitly forward them with (`secrets: inherit`) which is what I had to do in this [PR](https://github.com/consensus-shipyard/ipc/pull/674/files)

Reply:
I don't think the URL is a secret per se. Besides, it's often that a RPC URL fails or take too long to response. In that case I would just make a PR to switch it to a different provider. But if we keep track them in secrets, we probably already forgotten what URL we were using. That just make things more implicit for maintainers.